pkg/operator/kuryr.go

```diff
@@ -0,0 +1,147 @@
+
+	netv1 "github.com/openshift/api/network/v1"
+	"github.com/operator-framework/operator-sdk/pkg/util/k8sutil"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	uns "k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/openshift/cluster-network-operator/pkg/render"
+)
+
+// renderKuryr returns manifests for Kuryr SDN.
+// This creates
+// - the ClusterNetwork object
+// - the kuryr namespace
+// - the kuryr-controller deployment
+// - the kuryr-daemon daemonset
+// and some other small things.
+func (h *Handler) renderKuryr() ([]*uns.Unstructured, error) {
+	operConfig := h.config.Spec
+	c := operConfig.DefaultNetwork.KuryrConfig
+
+	objs := []*uns.Unstructured{}
+
+	// generate master network configuration
+	ippools := []netv1.ClusterNetworkEntry{}
+	for _, net := range operConfig.ClusterNetworks {
+		ippools = append(ippools, netv1.ClusterNetworkEntry{CIDR: net.CIDR, HostSubnetLength: net.HostSubnetLength})
+	}
+
+	clusterNet := netv1.ClusterNetwork{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "network.openshift.io/v1",
+			Kind:       "ClusterNetwork",
+		},
+		ObjectMeta: metav1.ObjectMeta{Name: netv1.ClusterNetworkDefault},
+
+		ServiceNetwork:  operConfig.ServiceNetwork,
+		PluginName:      sdnPluginName("OpenStack Kuryr-Kubernetes"),
+		ClusterNetworks: ippools,
+		//TODO(dulek): Do I need VXLANPort here?
+		// VXLANPort:       c.VXLANPort,
+
+		Network:          ippools[0].CIDR,
+		HostSubnetLength: ippools[0].HostSubnetLength,
+	}
+	obj, err := k8sutil.UnstructuredFromRuntimeObject(&clusterNet)
+	if err != nil {
+		// This is very unlikely
+		return nil, errors.Wrap(err, "failed to transmutate ClusterNetwork")
+	}
+	objs = append(objs, obj)
+
+	// render the manifests on disk
+	data := render.MakeRenderData()
+
+	// OpenStack credentials
+	data.Data["OpenStackCACertificate"] = c.OpenStackCACertificate
+	data.Data["OpenStackAuthURL"] = c.OpenStackCredentials.AuthURL
+	data.Data["OpenStackPassword"] = c.OpenStackCredentials.Password
+	data.Data["OpenStackProjectDomainName"] = c.OpenStackCredentials.ProjectDomainName
+	data.Data["OpenStackProjectId"] = c.OpenStackCredentials.ProjectId
+	data.Data["OpenStackUserDomainName"] = c.OpenStackCredentials.UserDomainName
+	data.Data["OpenStackUsername"] = c.OpenStackCredentials.Username
+	data.Data["KuryrProject"] = c.KuryrProject
+
+	data.Data["PodSubnet"] = c.PodSubnet
+	data.Data["PodSecurityGroups"] = c.PodSecurityGroups
+	data.Data["ServiceSubnet"] = c.ServiceSubnet
+	data.Data["WorkerNodesSubnet"] = c.WorkerNodesSubnet
+
+	// kuryr-daemon DaemonSet data
+	data.Data["DaemonEnableProbes"] = c.DaemonEnableProbes
+	data.Data["DaemonProbesPort"] = c.DaemonProbesPort
+	data.Data["CNIBinDir"] = c.CNIBinDir
+
+	// kuryr-controller Deployment data
+	data.Data["ControllerEnableProbes"] = c.ControllerEnableProbes
+	data.Data["ControllerProbesPort"] = c.ControllerProbesPort
+
+	// TODO(dulek): How do I get image names correctly?
+	data.Data["NodeImage"] = os.Getenv("NODE_IMAGE")
```

Comment:
AFAICT, this is the correct way to do node images. It would be your team's responsibility to build a pipeline for the kuryr images,.